scripts/migration/GPII-4014/verify.js

```diff
@@ -0,0 +1,163 @@
+
+// A sample command that runs this script in the universal root directory:
+// node scripts/migration/GPII-4014/verify.js http://localhost:25984 "clientCredential-nova1" "clientCredential-nova2"
+
+"use strict";
+
+var fluid = require("infusion"),
+    gpii = fluid.registerNamespace("gpii"),
+    url = require("url");
+
+fluid.registerNamespace("gpii.migration.GPII4014");
+
+require("./shared/migratedValues.js");
+require("../../shared/dbRequestUtils.js");
+require("../../../gpii/node_modules/gpii-db-operation/src/DbUtils.js");
+
+/**
+ * Create a set of options for this script.
+ * The options are based on the command line parameters and a set of database constants.
+ * @param {Array} processArgv - The command line arguments.
+ * @return {Object} - The options.
+ */
+gpii.migration.GPII4014.initOptions = function (processArgv) {
+    var options = {};
+    options.couchDbUrl = processArgv[2] + "/gpii";
+    options.novaClientCredentials = process.argv.splice(3);
+
+    // Set up database specific options
+    options.allDocsUrl = options.couchDbUrl + "/_all_docs?include_docs=true";
+    options.allDocs = [];
+    options.parsedCouchDbUrl = url.parse(options.couchDbUrl);
+    options.postOptions = {
+        hostname: options.parsedCouchDbUrl.hostname,
+        port: options.parsedCouchDbUrl.port,
+        path: "/gpii/_bulk_docs",
+        auth: options.parsedCouchDbUrl.auth,
+        method: "POST",
+        headers: {
+            "Accept": "application/json",
+            "Content-Length": 0,          // IMPORTANT: FILL IN PER REQUEST
+            "Content-Type": "application/json"
+        }
+    };
+    console.log("COUCHDB_URL: '" +
+        options.parsedCouchDbUrl.protocol + "//" +
+        options.parsedCouchDbUrl.hostname + ":" +
+        options.parsedCouchDbUrl.port +
+        options.parsedCouchDbUrl.pathname + "'"
+    );
+    return options;
+};
+
+/**
+ * Create the step that retrieves all documents from the database
+ * @param {Object} options - All docs URL and whether to filter:
+ * @param {Array} options.allDocsUrl - The url for retrieving all documents in the database.
+ * @return {Promise} - A promise that resolves retrieved documents.
+ */
+gpii.migration.GPII4014.verifyAllDocs = function (options) {
+    var details = {
+        requestUrl: options.allDocsUrl,
+        requestErrMsg: "Error retrieving documents from the database: ",
+        responseDataHandler: gpii.migration.GPII4014.verifyDocsData,
+        responseErrMsg: "Error retrieving documents from database: "
+    };
+    return gpii.dbRequest.configureStep(details, options);
+};
+
+/**
+ * Given all the documents from the database, update their data according to the new data model.
+ * @param {String} responseString - the response from the request to get all documents.
+ * @param {Object} options - Where to store the to-be-updated documents:
+ * @param {Array} options.allDocs - Accumulated documents.
```

Comment:
This doc comment doesn't seem right - the docs do not appear in the options, although there is the element novaClientCredentials which is undocumented, and the options receive totalNumOfDocs which it doesn't appear is used by the client